File: backend/src/main/java/com/mycompany/online_shop_backend/controller/OrderController.java
package com.mycompany.online_shop_backend.controller;

import com.mycompany.online_shop_backend.controller.model.request.OrderRequest;
import com.mycompany.online_shop_backend.controller.model.response.CreatedOrderResponse;
import com.mycompany.online_shop_backend.controller.model.response.UserOrderDto;
import com.mycompany.online_shop_backend.service.OrderService;
import com.mycompany.online_shop_backend.service.SecurityService;
import com.mycompany.online_shop_backend.service.model.OrderDto;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;

import javax.servlet.http.HttpServletRequest;
import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping(produces = MediaType.APPLICATION_JSON_VALUE)
public class OrderController {

    private final OrderService orderService;
    private final SecurityService securityService;

    @PostMapping(path = "/api/v1/orders", consumes = MediaType.APPLICATION_JSON_VALUE)
    @ResponseStatus(HttpStatus.CREATED)
    public CreatedOrderResponse createOrder(@RequestBody OrderRequest request) {
        OrderDto orderDto = orderService.save(request);
        return CreatedOrderResponse.toDto(orderDto);
    }

    @GetMapping(path = "/api/v1/users/{id}/orders")
    public List<UserOrderDto> getUserOrders(HttpServletRequest request) {
        String email = securityService.getUsernameFromRequest(request);
        List<OrderDto> orderDtos = orderService.getOrdersByEmail(email);
        return orderDtos.stream().map(UserOrderDto::toDto).toList();
    }
}

File: order-service/src/main/kotlin/com/example/orderservice/api/rest/OrderController.kt
package com.example.orderservice.api.rest

import com.example.orderservice.api.rest.model.CreateOrderRequest
import com.example.orderservice.api.rest.model.Order
import com.example.orderservice.service.OrderService
import org.springframework.data.web.PagedModel
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.*
import java.util.*

@RestController
@RequestMapping("/api/v1", produces = [MediaType.APPLICATION_JSON_VALUE])
class OrderController(private val orderService: OrderService) {

    @PostMapping("/orders/search")
    fun orders(orderRequestParams: OrderRequestParams): PagedModel<Order> =
        orderService.getOrders(orderRequestParams)

    @GetMapping("/orders/{orderId}")
    fun orderById(@PathVariable("orderId") orderId: UUID): Order = orderService.getOrderById(orderId)

    @PostMapping("/orders", consumes = [MediaType.APPLICATION_JSON_VALUE])
    fun createOrder(
        @RequestHeader(IDEMPOTENCY_KEY) idempotencyKey: UUID,
        @RequestBody request: CreateOrderRequest
    ): Order = orderService.createOrder(idempotencyKey, request)

}